leakcanary-android-core/src/main/java/leakcanary/internal/activity/screen/LeakTraceWordWrapper.kt

```diff
@@ -0,0 +1,368 @@
+package leakcanary.internal.activity.screen
+
+import java.util.Stack
+
+/**
+ * Performs word wrapping of leak traces.
+ */
+object LeakTraceWordWrapper {
+
+  private const val SPACE_CHARACTER = '\u0020'
+  private const val TILDE_CHARACTER = '\u007E'
+  private const val PERIOD_CHARACTER = '\u002E'
+  private const val BREAKING_CHARACTER_LENGTH = 1
+  const val BREAK_CHARACTER = '\n'
+
+  /**
+   * This implements a greedy wrapping algorithm.
+   *
+   * <p>
+   *
+   * Each line is split into words, using spaces and dots as separators. From there, a wrapped
+   * version of the line is built by adding one word after the other until the string doesn't fit
```

Comment:
Instead of splitting the entire line by words and adding until the threshold is reached, have you considered walking back from the character at WIDTH until you find a space or period?

Reply:
Yep that's a good idea. I will try that.